tietokanta/src/main/resources/db/migration/R___Materiaalin_kaytto.sql

```diff
@@ -53,62 +53,14 @@ END;
 $$ LANGUAGE plpgsql;
 
 
--- Päivitä kaikki materiaalin käyttö päivämäärälle
-CREATE OR REPLACE FUNCTION paivita_materiaalin_kaytto_hoitoluokittain_paivalle(pvm_ DATE)
-  RETURNS VOID AS $$
-DECLARE
-  rivi RECORD;
-BEGIN
-  DELETE FROM urakan_materiaalin_kaytto_hoitoluokittain WHERE pvm = pvm_;
-  FOR rivi IN SELECT t.urakka, rp.talvihoitoluokka, mat.materiaalikoodi,
-                     sum(mat.maara) as maara
-      	        FROM toteuma t
-                     JOIN toteuman_reittipisteet tr ON tr.toteuma = t.id
-                     JOIN LATERAL unnest(tr.reittipisteet) rp ON true
-                     JOIN LATERAL unnest(rp.materiaalit) mat ON true
-               WHERE t.alkanut BETWEEN pvm_ AND (pvm_ + interval '1 day')::DATE
-                 AND t.poistettu IS FALSE
-            GROUP BY t.urakka, rp.talvihoitoluokka, mat.materiaalikoodi
-                     -- Tässä otetaan talteen erikoiskäsittelyllä kaikki käsin syötetyt toteumat, ja annetaan niille
-                     -- hoitoluokaksi 100 eli 'ei tiedossa', ks. esim VHAR-4204
-   UNION
-  SELECT t.urakka, 100 AS talvihoitoluokka, tm.materiaalikoodi,
-         sum(tm.maara) as maara
-    FROM toteuma t
-             JOIN toteuma_materiaali tm ON tm.toteuma = t.id and tm.poistettu IS FALSE
-   WHERE t.lahde = 'harja-ui' and
-         t.alkanut BETWEEN pvm_ AND (pvm_ + interval '1 day')::DATE
-     AND t.poistettu IS FALSE
-   GROUP BY t.urakka, talvihoitoluokka, tm.materiaalikoodi, t.alkanut::DATE
-  LOOP
-    INSERT INTO urakan_materiaalin_kaytto_hoitoluokittain
-                (pvm, materiaalikoodi, talvihoitoluokka, urakka, maara)
-         VALUES (pvm_,
-	         rivi.materiaalikoodi,
-		 COALESCE(rivi.talvihoitoluokka, 100),
-		 rivi.urakka,
-		 rivi.maara)
-    ON CONFLICT ON CONSTRAINT uniikki_urakan_materiaalin_kaytto_hoitoluokittain DO
-    UPDATE SET maara = urakan_materiaalin_kaytto_hoitoluokittain.maara + EXCLUDED.maara;
-  END LOOP;
-  RETURN;
-END;
-$$ LANGUAGE plpgsql;
-
+-- Päivitä kaikki materiaalin käyttö päivämäärävälille. Huom! Tätä ei tällä hetkellä käytetä sovelluslogiikassa,
+-- mutta tämä on voimakas työkalu jos tarvii päivitellä tuotannossa datoja. Käytä harkiten.
 CREATE OR REPLACE FUNCTION paivita_materiaalin_kaytto_hoitoluokittain_aikavalille(alku DATE, loppu DATE)
   RETURNS VOID AS $$
 DECLARE
-  pvm DATE;
 BEGIN
-  pvm := alku;
-  LOOP
-    IF pvm > loppu THEN
-      EXIT;
-    END IF;
-    -- RAISE NOTICE 'Päivitetään materiaalin käyttö hoitoluokittain: %', pvm;
-    PERFORM paivita_materiaalin_kaytto_hoitoluokittain_paivalle(pvm);
-    pvm := pvm + 1;
- END LOOP;
- RETURN;
+    -- Tässä passataan urakka-parametriksi NULL, jolloin päivitetään kaikki urakat ko. aikaväliltä
+    PERFORM paivita_urakan_materiaalin_kaytto_hoitoluokittain(NULL, alku, loppu);
```

Comment:
Kun urakkaa ei passata, niin deletointi jää tekemättä:
DELETE FROM urakan_materiaalin_kaytto_hoitoluokittain
  WHERE pvm BETWEEN alkupvm AND (loppupvm + interval '1 day')::DATE
        AND urakka = u;
Haittaako se mitään?

Reply:
ah. onhan se tuolla. Ei haittaa mitään. Pahoittelut. En hoksannut, että muutit sitäkin tarkistusta. Sori